### PR TITLE
Implement configurable Smeagol link in primary nav 

### PR DESCRIPTION
**Pull request by schnatterer**
## Proposed changes

Resolves feature request from scm-manager/website#16.

Picks up original idea from schnatterer/scm-smeagol-plugin@14378cc, integrates it into the official scm-smeagol-plugin and adds a configuration option.

image

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described and the description can be used as commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog`
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: src/main/js/SmeagolConfigurationForm.tsx [hunk 1]
```diff
@@ -70,6 +71,13 @@ const SmeagolConfigurationForm: FC<Props> = ({ initialConfiguration, onConfigura
         checked={changedConfig.enabled}
         onChange={valueChangeHandler}
       />
+      <Checkbox
+        name="navLinkEnabled"
+        label={t("scm-smeagol-plugin.form.navLinkEnabled")}
+        helpText={t("scm-smeagol-plugin.form.navLinkEnabledHelp")}
+        checked={changedConfig.navLinkEnabled}
+        onChange={valueChangeHandler}
+      />
```
Comment (pfeuffer) [inline src/main/js/SmeagolConfigurationForm.tsx:80]: The nav link option should be disabled as long as the whole configuration is not enabled.

```suggestion
      <Checkbox
        name="navLinkEnabled"
        label={t("scm-smeagol-plugin.form.navLinkEnabled")}
        helpText={t("scm-smeagol-plugin.form.navLinkEnabledHelp")}
        checked={changedConfig.navLinkEnabled}
        onChange={valueChangeHandler}
        disabled={!changedConfig.enabled}
      />
```

File: src/main/java/com/cloudogu/smeagol/IndexLinkEnricher.java [hunk 2]
```diff
@@ -38,17 +38,23 @@
...
   public void enrich(HalEnricherContext context, HalAppender appender) {
-    appender.linkArrayBuilder("smeagol")
-      .append("repositories", smeagolLinkBuilder.getRepositoriesLink())
-      .build();
+    HalAppender.LinkArrayBuilder smeagolArray = appender.linkArrayBuilder("smeagol");
+    smeagolArray
+      .append("repositories", smeagolLinkBuilder.getRepositoriesLink());
+    if (configuration.get().isNavLinkEnabled()) {
+      smeagolArray.append("smeagolRoot", configuration.get().getSmeagolUrl());
+    }
```
Comment (pfeuffer) [inline src/main/java/com/cloudogu/smeagol/IndexLinkEnricher.java:56]: This link should only be available, if the whole plugin is enabled:

```suggestion
    if (configuration.get().isEnabled() && configuration.get().isNavLinkEnabled()) {
      smeagolArray.append("smeagolRoot", configuration.get().getSmeagolUrl());
    }
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+### Added
+- Implement configurable Smeagol link in primary nav  ([#2](https://github.com/scm-manager/scm-smeagol-plugin/pull/2))
```
Comment (pfeuffer) [inline CHANGELOG.md:9]: ;-)

```suggestion
- Implement configurable Smeagol link in primary navigation  ([#2](https://github.com/scm-manager/scm-smeagol-plugin/pull/2))
```

**Review by pfeuffer — CHANGES_REQUESTED** [review]
Thanks for this PR. Just small remarks from my side.

Welcome back to the team :-)

**Comment by schnatterer** [comment]
Thanks @pfeuffer. Glad to be back for a short visit :wink: 
Changes LGTM.

File: src/main/resources/locales/en/plugins.json [hunk 4]
```diff
@@ -5,6 +5,8 @@
       "header": "Global Smeagol Configuration",
       "enabled": "Enabled",
       "enabledHelp": "If \"Enabled\" SCM-Manager creates a link in the navigation for repositories with a Smeagol wiki that leads to this wiki.",
+      "navLinkEnabled": "Primary Navigation Link Enabled",
+      "navLinkEnabledHelp": "Wenn \"Enabled\" SCM-Manager creates a link in the primary navigation bar that leads to the home page of Smeagol. Changes can only be seen after a page reload.",
```
Comment (pfeuffer) [inline src/main/resources/locales/en/plugins.json:9]: Sorry, did not see this one...
```suggestion
      "navLinkEnabledHelp": "If \"Enabled\" SCM-Manager creates a link in the primary navigation bar that leads to the home page of Smeagol. Changes can only be seen after a page reload.",
```

**Review by pfeuffer — CHANGES_REQUESTED** [review]
Sorry, missed this little one

**Pull request merged**